resources/user.rb

```diff
@@ -0,0 +1,53 @@
+#
+# Cookbook Name:: sensu-go
+# Resource:: agent
+#
+# Copyright:: 2018 Sensu, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+resource_name :sensu_user
+
+action_class do
+  include SensuCookbook::Helpers
+end
+
+property :config_home, String, default: '/etc/sensu'
+
+property :password, String, required: true
+property :roles, Array
+property :disabled, [true, false], default: false
+
+action :create do
+  directory object_dir do
+    action :create
+    recursive true
+  end
+
+  file object_file do
```

Comment:
this file contains a password so we should use `sensitive: true` to prevent it from being displayed in the logs.

Reply:
Also after seeing the intended implementation I am not sure this is the best option file or no file as this means that all users passwords are stored on the server in plain text. Lets see if we can hop on a call and hash out some stuff. I will DM you in slack to figure out a time that works.

Reply:
@majormoses 👍 just caught up on the slack chat - I think what I'd like to do is remove the user resource from this PR for now, as there are still open questions, and resubmit in another PR that just adds RBAC resource support (there are one or two more things to add in addition to users) once we've hashed out an appropriate way to do this. I'll try to get that done at some point today.

Reply:
@mercul3s I had proposed the same but @oferrigni wanted to see if we could get this done here I am fine with removing the `user` resource for now.